mempool/bench_test.go

```diff
@@ -1,24 +1,85 @@
 package mempool
 
 import (
+	"crypto/rand"
 	"encoding/binary"
+	"fmt"
+	"math/big"
+	"strconv"
+	"strings"
 	"testing"
 
-	"github.com/tendermint/tendermint/abci/example/kvstore"
+	"github.com/tendermint/tendermint/abcix/example/kvstore"
 	cfg "github.com/tendermint/tendermint/config"
 	"github.com/tendermint/tendermint/proxy"
+	"github.com/tendermint/tendermint/types"
 )
 
-func BenchmarkCheckTx(b *testing.B) {
+// BenchmarkClistCheckTx-8   	     768	   1510694 ns/op
+func BenchmarkClistCheckTx(b *testing.B) {
+	benchmarkCheckTx(b, enumclistmempool)
+}
+
+// BenchmarkLLRBCheckTx-8   	     646	   1573239 ns/op
+func BenchmarkLLRBCheckTx(b *testing.B) {
+	benchmarkCheckTx(b, enumllrbmempool)
+}
+
+func benchmarkCheckTx(b *testing.B, enum mpEnum) {
 	app := kvstore.NewApplication()
-	cc := proxy.NewLegacyLocalClientCreator(app)
-	mempool, cleanup := newLegacyMempoolWithAppAndConfig(cc, cfg.ResetTestRoot("mempool_test"), enumclistmempool)
+	cc := proxy.NewLocalClientCreator(app)
+	mempool, cleanup := newMempoolWithAppAndConfig(cc, cfg.ResetTestRoot(fmt.Sprintf("mempool_test_%d", enum)), enum)
 	defer cleanup()
+	size := 100
+	txs := types.Txs{}
+	for i := 0; i < size; i++ {
+		txBytes := make([]byte, 20)
+		priority := strconv.FormatInt(int64(i)%100, 10)
+		tx := "k" + strconv.Itoa(i) + "=v" + strconv.Itoa(i) + ","
+		extra := 20 - len(tx) - len(priority) - 1
+		tx = tx + strings.Repeat("f", extra) + "," + priority
+		copy(txBytes, tx)
+		txs = append(txs, txBytes)
+	}
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		for j := 0; j < size; j++ {
+			mempool.CheckTx(txs[j], nil, TxInfo{})
+		}
+		b.StopTimer()
+		mempool.Flush()
```

Comment:
this change the semantics of this benchmark because it will delete all txs. why do so?

Reply:
In my consideration, if the mempool didn't flush after first-round CheckTx, then all following CheckTx will insert duplicated txs, does it make sense?